src/oas.js

```diff
@@ -195,20 +225,22 @@ class Oas {
     return new Operation(this, path, method, operation);
   }
 
-  findOperation(url, method) {
-    const { origin, pathname } = new URL(url);
+  findOperation(url, method, index = 0) {
+    const { origin } = new URL(url);
+    const originRegExp = new RegExp(origin);
     const { servers, paths } = this;
 
-    const targetServer = servers.find(s => s.url === origin);
+    const targetServer = servers.find(s => originRegExp.exec(s.url));
     if (!targetServer) return undefined;
 
-    const annotatedPaths = generatePathMatches(paths, pathname, origin);
+    const [, pathName] = url.split(targetServer.url);
+    const annotatedPaths = generatePathMatches(paths, pathName, targetServer.url);
     if (!annotatedPaths.length) return undefined;
 
     const includesMethod = filterPathMethods(annotatedPaths, method);
     if (!includesMethod.length) return undefined;
 
-    return findTargetPath(includesMethod);
+    return findTargetPath(includesMethod, index);
```

Comment:
`findOperation` should probably pull back the spec of the operation instead.

Reply:
Instead of returning the index and handling it in the front end, return the whole OAS file in this payload?

Reply:
I was originally thinking a few weeks ago that running `findOperation('/pets/1`, 'get')` would return back everything in the OAS under `paths[/pets].get` and the FE could use that.

Reply:
It is doing that already! Per our discussion I've migrated the index to our main repo.